derive-input/src/internal/dsl/method.rs

```diff
@@ -1360,12 +1390,34 @@ pub(in crate::internal) fn for_method(
                                 });
                             };
 
+                            let hook_call = if spacetimedsl_table.hook_on_delete {
+                                let hook_fn_name = format_ident!("on_{}_delete", singular_table_name);
+                                quote! {
+                                    // Collect rows before deletion to call hooks on them
+                                    // We need to collect because filter_map would consume the rows,
+                                    // but we need to keep them to call hooks before actual deletion
+                                    let rows_to_delete: Vec<_> = primary_key_values_of_rows_to_delete.iter().filter_map(|pk| {
+                                        self.ctx().db().#singular_table_name().#primary_key_column_name().find(*pk)
+                                    }).collect();
+                                    
+                                    for row in &rows_to_delete {
+                                        if let Err(error_msg) = #hook_fn_name(self, row) {
+                                            return Err(spacetimedsl::SpacetimeDSLError::Error(error_msg));
+                                        }
```

Comment:
hook functions should return `spacetimedsl::SpacetimeDSLError` instead of string.